data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSourceSchemaFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.util;
+
+import org.opensearch.dataprepper.plugins.kafka.consumer.AvroConsumer;
+import org.opensearch.dataprepper.plugins.kafka.consumer.JsonConsumer;
+import org.opensearch.dataprepper.plugins.kafka.consumer.KafkaSourceSchemaConsumer;
+import org.opensearch.dataprepper.plugins.kafka.consumer.PlainTextConsumer;
+
+/**
+ * This is a factory class which will return instance of AVRO,JSON or PLAINTEXT
+ * class based on the schema type configured in the pipelines.yaml
+ */
+public class KafkaSourceSchemaFactory {
+  KafkaSourceSchemaFactory() {
+  }
+
+  @SuppressWarnings("rawtypes")
+  public static KafkaSourceSchemaConsumer getSchemaType(MessageFormat format) {
```

Comment:
name of method does not reflect the implementation. createConsumer() could be one alternative.

Reply:
Resolved. changed the method name to createConsumer()